src/symbolize/mod.rs

```diff
@@ -188,6 +196,26 @@ impl<'a> SymbolName<'a> {
     }
 }
 
+// With std enabled attempts to lossy convert bytes, on core uses
+// strict checking
+#[cfg(feature = "std")]
+fn format_symbol_name(fmt: fn(&str, &mut fmt::Formatter) -> fmt::Result,
+                      bytes: &[u8],
+                      f: &mut fmt::Formatter)
+    -> fmt::Result
+{
+    fmt(&*String::from_utf8_lossy(bytes), f)
+}
+
+#[cfg(not(feature = "std"))]
+fn format_symbol_name(fmt: fn(&str, &mut fmt::Formatter) -> fmt::Result,
+                      bytes: &[u8],
+                      f: &mut fmt::Formatter)
+    -> fmt::Result
+{
+    fmt(str::from_utf8(bytes).unwrap(), f)
```

Comment:
I don't think we'll want to implement this here with `unwrap()` for core, we'll want the same logic for both std/core which is to attempt to interpret as utf-8 but otherwise never panic.

(for example panicking definitely means we can't integrate this later into libstd